### PR TITLE
[SYCL] Update tests for disabling fallback assert feature on FPGA device and disable Windows assert tests on CUDA

### PR DESCRIPTION
**Pull request by s-kanaev**
Tests update for https://github.com/intel/llvm/pull/4780 and https://github.com/intel/llvm/issues/4810

### CODE REVIEW

**Review by vladimirlaz — DISMISSED** [review]

**Comment by bader** [comment]
I think it's worth to fix https://github.com/intel/llvm/issues/4810 as well, while you are here.

File: SYCL/Assert/assert_in_kernels.cpp [hunk 1]
```diff
@@ -6,13 +6,17 @@
 // RUN: %CPU_RUN_PLACEHOLDER FileCheck %s --input-file %t.txt
 // RUN: %GPU_RUN_PLACEHOLDER %t.out &> %t.txt || true
 // RUN: %GPU_RUN_PLACEHOLDER FileCheck %s --input-file %t.txt
-// RUN: %ACC_RUN_PLACEHOLDER %t.out &> %t.txt || true
-// RUN: %ACC_RUN_PLACEHOLDER FileCheck %s --input-file %t.txt
+// Shouldn't fail on ACC as fallback assert isn't enqueued there
+// RUN: %ACC_RUN_PLACEHOLDER %t.out &> %t.txt
+// RUN: %ACC_RUN_PLACEHOLDER FileCheck %s --check-prefix=CHECK_ACC --input-file %t.txt
```
Comment (bader) [inline SYCL/Assert/assert_in_kernels.cpp]:
```suggestion
// RUN: %ACC_RUN_PLACEHOLDER FileCheck %s --check-prefix=CHECK-ACC --input-file %t.txt
```

Comment (s-kanaev) [reply SYCL/Assert/assert_in_kernels.cpp]: Done.

**Comment by s-kanaev** [comment]
@bader , @vladimirlaz , @romanovvlad , a friendly ping. The tests are running OK with appropriate changes in intel/llvm#4780.

File: SYCL/.clang-format [hunk 2]
```diff
@@ -1,2 +1,2 @@
 BasedOnStyle: LLVM
-CommentPragmas: "(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK) *:|expected-"
+CommentPragmas: "(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK|CHECK[A-Za-z0-9_-]+) *:|expected-"
```
Comment (bader) [inline SYCL/.clang-format]:
```suggestion
CommentPragmas: "(RUN|FAIL|REQUIRES|UNSUPPORTED|CHECK[A-Za-z0-9_-]*) *:|expected-"
```
Will this work?

**Review by bader — DISMISSED** [review]

File: SYCL/.clang-format (hunk 2 above)
Comment (s-kanaev) [reply SYCL/.clang-format]: I guess so. Lets try it out.

**Review by romanovvlad — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**